admin_app.py

```diff
@@ -25,59 +28,90 @@ def make_secret_key():
+    client_kwargs={
+        'scope': 'openid email profile',
+    },
+)
 
 
-def identity():
-    # Get email from IAP JWT identity. This validates the token.
-    project_number = os.getenv('PROJECT_NUMBER')
-    project_id = os.getenv('PROJECT_ID')
-    iap_jwt = request.headers.get('X-Goog-IAP-JWT-Assertion')
-    identity = validate_iap_jwt_from_app_engine(iap_jwt, project_number,
-                                                project_id)
-    user_email = identity[1]
-    return user_email
+@app.route('/')
+def index():
+    if 'profile' in session:
+        return redirect('/admin/client/?flt0_0=1')
+    else:
+        return redirect('/login')
 
 
-def generate_code(client):
-    """ Generate OAO Standard Client code value from
-        - First 2 letters of assigned account name
-        - Start year
-        - sum of unicode values of org name, modulo 999, zero-padded
-    """
-    if client.client_organization_code:
-        # no-op if client already has a code
-        return client.client_organization_code
-    norm_name = client.assigned_account_name.upper().strip().replace(
-        'THE ', '')
-    abbrev_name = norm_name[:2]
+@app.route('/callback')
+def callback_handling():
     try:
-        start_year = client.contract_start_date.year
-    except AttributeError:
-        start_year = datetime.now().year
-    suffix = str(
-        sum(bytearray(client.client_organization_name, 'utf-8')) %
-        999).zfill(3)
-    client_code = '{0}{1}-{2}'.format(abbrev_name, start_year, suffix)
-    return client_code
+        auth0.authorize_access_token()
+    except BadRequestKeyError:
+        return ('<h1>Access denied</h1>'
+                '<p>Please <a href="/login">login</a> '
+                'with a valid adops.com account</p>'), 400
+    resp = auth0.get('userinfo')
+    userinfo = resp.json()
+
+    session['jwt_payload'] = userinfo
+    session['profile'] = {
+        'user_id': userinfo['sub'],
+        'name': userinfo['name'],
+        'email': userinfo['email']
+    }
+    return redirect('/admin/client/?flt0_0=1')
 
 
-@app.route('/')
-def index():
-    user_email = identity()
-    try:
-        greeting_name = user_email.split('.')[0].title()
-    except AttributeError:
-        greeting_name = 'neighbor'
-    return '<a href="/admin/client/?flt0_0=1">Admin Ahoy</a>, {}!'.format(
-        greeting_name)
+@app.route('/login')
+def login():
+    return auth0.authorize_redirect(
+        redirect_uri=AUTH0_CALLBACK_URL, audience=AUTH0_AUDIENCE)
 
 
-"""
-Helper functions for form choices
-TODO: Pull out to own file (or add to __init__?)
+@app.route('/logout')
+def logout():
+    session.clear()
+    params = {
+        'returnTo': url_for('index', _external=True, _scheme='https'),
+        'client_id': AUTH0_CLIENT_ID
+    }
+    return redirect(auth0.api_base_url + '/v2/logout?' + urlencode(params))
+
+
+class AuthMixin():
+    def is_accessible(self):
+        if 'profile' in session:
```

Comment:
Am I missing something or is this allowing any 0Auth2 account in? Is there future work planned to specify particular accounts that are and are not allowed?

I feel like I'm missing something

Reply:
I think that might be specified elsewhere in auth0 config 🤔

Reply:
@jessebmiller Access control is set up in the Auth0 dashboard. Currently there is a rule limiting access to @adops.com accounts, and once in users have full edit privileges. But this can be adjusted to finer grain as needed. For API access, Auth0 provides a UI screen. Interactive apps like this one use rules: https://auth0.com/docs/rules/current

Reply:
Also, now that I think of it, if we want to go all in on using their service, we should probably version-control the access rules. Setting up rules via the Management API instead of the dashboard would help facilitate this. I'll add an issue.